### PR TITLE
Add a new package for AMG and fix build issues in the amg2013 package

### PR DESCRIPTION
**Pull request by bhatele**

### CODE REVIEW

**Comment by adamjstewart** [comment]
These should probably be merged into a single package. I'll let you decide the best name for the package.

**Comment by bhatele** [comment]
@adamjstewart They are actually different code bases and do different things. So in my opinion, they should remain separate.

File: var/spack/repos/builtin/packages/amg2013/package.py [hunk 1]
```diff
@@ -40,24 +40,28 @@ class Amg2013(MakefilePackage):
             url='https://codesign.llnl.gov/amg2013/amg2013.tgz')
 
     variant('openmp', default=True, description='Build with OpenMP support')
-    variant('assumepartition', default=False, description='Assumed partition (for thousands of processors)')
+    variant('assumedpartition', default=False, description='Use assumed partition (for thousands of processors)')
+    variant('64bitglobals', default=False, description='Use 64-bit integers for global variables')
```
Comment (davydden) [inline var/spack/repos/builtin/packages/amg2013/package.py]: please call it `int64` to be consistent with other packages (superlu-dist, mumps, hypre, petsc, etc)

File: var/spack/repos/builtin/packages/amg/package.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    homepage = "https://codesign.llnl.gov/amg2013.php"
+    git      = "https://github.com/LLNL/AMG"
+
+    version('1.0', git=git, tag='1.0')
+    version('develop', git=git, branch='master')
+
+    variant('openmp', default=True, description='Build with OpenMP support')
+    variant('opt', default=False, description='Additional optimizations')
+    variant('64bitglobals', default=False, description='Use 64-bit integers for global variables')
```
Comment (davydden) [inline var/spack/repos/builtin/packages/amg/package.py]: same here, `int64`

File: var/spack/repos/builtin/packages/amg/package.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    tags = ['proxy-app', 'ecp-proxy-app']
+
+    homepage = "https://codesign.llnl.gov/amg2013.php"
+    git      = "https://github.com/LLNL/AMG"
+
+    version('1.0', git=git, tag='1.0')
+    version('develop', git=git, branch='master')
+
+    variant('openmp', default=True, description='Build with OpenMP support')
+    variant('opt', default=False, description='Additional optimizations')
```
Comment (davydden) [inline var/spack/repos/builtin/packages/amg/package.py]: we dont have convection for those, but in `dealii` i called this `optflags`

**Comment by bhatele** [comment]
@davydden All done.

**Pull request merged**